src/cascadia/TerminalApp/AppActionHandlers.cpp

```diff
@@ -1021,6 +1021,42 @@ namespace winrt::TerminalApp::implementation
         args.Handled(true);
     }
 
+    void TerminalPage::_HandleSearchForText(const IInspectable& /*sender*/,
+                                            const ActionEventArgs& args)
+    {
+        if (args)
+        {
+            if (const auto& realArgs = args.ActionArgs().try_as<SearchForTextArgs>())
+            {
+                if (const auto termControl{ _GetActiveControl() })
+                {
+                    if (termControl.HasSelection())
+                    {
+                        const auto selections{ termControl.SelectedText(true) };
+
+                        // concatenate the selection into a single line
+                        auto searchText = std::accumulate(selections.begin(), selections.end(), std::wstring());
+
+                        // make it compact by replacing consecutive whitespaces with a single space
+                        searchText = std::regex_replace(searchText, std::wregex(LR"(\s+)"), L" ");
+
+                        if (realArgs.WrapWithQuotes())

```

Comment:
alright, so I've got an idea here. It kills two of my concerns!

When you set up a search engine shortcut in ~~Firefox~~ Microsoft Edge (:wink:) it gives you a special string, `%s`, that you can insert into the query.

<img width="320" alt="image" src="https://github.com/microsoft/terminal/assets/189190/c5bfce28-94fd-4d76-9d8d-4875686fb119">

If we use `%s` just like them, we get some benefits:

1. we don't have to append, so people can put the query argument anywhere
2. we don't need `wrapWithQuotes`, because people can put quotes directly in the URL!